frontend/scripts/react-components/dashboard-element/dashboard-widget/dashboard-widget.container.jsx

```diff
@@ -1,129 +1,54 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
-import qs from 'qs';
-import pickBy from 'lodash/pickBy';
 import sortBy from 'lodash/sortBy';
-import DashboardWidget from 'react-components/dashboard-element/dashboard-widget/dashboard-widget.component';
+import DashboardWidgetComponent from 'react-components/dashboard-element/dashboard-widget/dashboard-widget.component';
 import DashboardWidgetTooltip from 'react-components/dashboard-element/dashboard-widget/dashboard-widget-tooltip';
-import Widget from 'react-components/widgets/widget.component';
-import CHART_CONFIG from 'react-components/dashboard-element/dashboard-widget/dashboard-widget-config';
+import { getConfig } from 'react-components/dashboard-element/dashboard-widget/dashboard-widget.selectors';
 
-class DashboardWidgetContainer extends Component {
-  static getGroupedY(meta) {
-    const { xAxis, yAxis, x, ...groupedY } = meta;
-    return groupedY;
-  }
-
-  static sortGroupedY(keys) {
-    return sortBy(Object.keys(keys), key => parseInt(key.substr(1), 10));
-  }
-
-  static getYKeys(meta, { yKeys, yKeysAttributes, colors }) {
-    if (!meta || !yKeys) return yKeys;
-    const groupedY = DashboardWidgetContainer.getGroupedY(meta);
-    return Object.keys(yKeys).reduce((yKeysTypesAcc, nextYKeyType) => {
-      const yKeyTypeAttributes = yKeysAttributes && yKeysAttributes[nextYKeyType];
-      return {
-        ...yKeysTypesAcc,
-        [nextYKeyType]: DashboardWidgetContainer.sortGroupedY(groupedY).reduce(
-          (groupedYKeysAcc, nextGroupedYKey, index) => ({
-            ...groupedYKeysAcc,
-            [nextGroupedYKey]: {
-              ...yKeyTypeAttributes,
-              fill: colors && colors[index],
-              stroke: colors && colors[index]
-            }
-          }),
-          {}
-        )
-      };
-    }, {});
-  }
-
-  static getColors(meta, { colors }, chartType, data) {
-    if (!meta) return colors;
-    if (chartType !== 'pie') {
-      const groupedY = DashboardWidgetContainer.getGroupedY(meta);
-      return DashboardWidgetContainer.sortGroupedY(groupedY).map((key, index) => ({
-        label: groupedY[key].label || meta.yAxis.label,
-        color: colors[index]
-      }));
-    }
-    return data.map((item, index) => ({
-      label: item.x,
-      color: colors[index]
-    }));
-  }
+const mapStateToProps = (state, { chartType, meta, data }) => ({
+  config: getConfig(state, { chartType, meta, data })
```

Comment:
I'm thinking this won't work as expected, because we're painting more than one widget per page, the memoization a widget produces gets busted by the next widget. We need to use a factory.

https://github.com/reduxjs/reselect#sharing-selectors-with-props-across-multiple-component-instances